account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py

```diff
@@ -56,13 +57,10 @@ def _create_move_and_lines(
             self, amount, debit_account_id, credit_account_id,
```

Comment:
@fclementic2c 

Why do we have to write that many Journal Entries. They can be created in only one Journal Entry having several Journal Items. It is more convenient  to have only one Journal Entry per Revaluation call.

Regarding Revaluation Entries for Bank/Liquidity accounts. Why to create that many Lines for each record found. 

They can be grouped by account / by currency / by partners and not using partner at all for creating the journal items. It is better to just group them (Liquidity accounts) by account / by currency.

What do you think?
Regards

Reply:
I agree - except for debtors and payable.
- Debtors and payable accounts must be grouped this way: 'account to revaluate+partner+currency' (and the journal entry must be posted this way too I mean)
- Others accounts must be grouped this way: 'account to revaluate+currency' 

Keep in mind that if accountants cannot precisely justify revaluated amounts they will reject the feature.

----------
For the record, this module is rather old, it was initially required by rather big company (Canonical) because they used to invoice everywhere in world and propose systematically 3 possible currencies to get paid (what ever the currency on the customer invoice). We wanted to calculate currency revaluation on debtors at the time.
Revaluation on bank account went later but we wanted to use the same code for this.
Finally, they ask for full details report and for this decided not to group revaluation entries but rather to create one entry per 'account to revaluate+partner+currency' -> this way the report was very simple just displaying move lines... but it is ugly on liquidity account.
Note also that in Uk you can do currency reval on any type of account.
----------

Reply:
@goliveirab 

- Let us make a refactor to create just one `Journal Entry` per Currency Revaluation run. 
- For (Receivables and Payables)  and their counterparts `Journal Items` will be create one per `account_id x partner_id x currency_id`.
- For (Liquidity + Other accounts set as Allow Revaluation) and their counterparts `Journal Items` will be created on per `account_id x currency_id` alone.

@fclementic2c we are on the same page regarding this:
> - Debtors and payable accounts must be grouped this way: 'account to revaluate+partner+currency' (and the journal entry must be posted this way too I mean)
> - Others accounts must be grouped this way: 'account to revaluate+currency'

Reply:
@hbto 

Could you please check the last change?

Thanks

Reply:
👍